packages/utils/src/lib/log-results.ts

```diff
@@ -0,0 +1,53 @@
+type PromiseFulfilledResult<T> = {
+  status: 'fulfilled';
+  value: T;
+};
+
+export function logMultipleResults<T>(
+  results: PromiseSettledResult<T>[],
+  messagePrefix: string,
+  succeededCallback?: (result: PromiseFulfilledResult<T>) => void,
+  failedCallback?: (result: PromiseRejectedResult) => void,
+) {
+  if (succeededCallback) {
+    const succeededResults = results.filter(
+      (result): result is PromiseFulfilledResult<T> =>
+        result.status === 'fulfilled',
+    );
```

Comment:
I think it would be convenient to create a type guard in `utils`:

```ts
export function isPromiseFulfilledResult<T>(
  result: PromiseSettledResult<T>,
): result is PromiseFulfilledResult<T> {
  return result.status === 'fulfilled';
}
```

Then we can reuse whenever we have `Array.filter` on a `PromiseSettledResult<T>[]`:

```suggestion
    const succeededResults = results.filter(isPromiseFulfilledResult);
```

Reply:
Done